logic/src/commonMain/kotlin/com/wire/kalium/logic/data/call/CallRepository.kt

```diff
@@ -139,124 +138,227 @@ internal class CallDataSource(
 
-        callProfile[conversationId]?.let { call ->
+        callMetadataProfile.data[conversationId]?.let { call ->
             callingLogger.i("updateCallParticipants() - conversationId: $conversationId with size of: ${participants.size}")
 
-            val updatedCalls = callProfile.calls.toMutableMap().apply {
+            val updatedCallMetadata = callMetadataProfile.data.toMutableMap().apply {
                 this[conversationId] = call.copy(
                     participants = participants,
                     maxParticipants = max(call.maxParticipants, participants.size + 1)
                 )
             }
 
-            _callProfile.value = callProfile.copy(
-                calls = updatedCalls
+            _callMetadataProfile.value = callMetadataProfile.copy(
+                data = updatedCallMetadata
             )
         }
     }
 
     override fun updateParticipantsActiveSpeaker(conversationId: String, activeSpeakers: CallActiveSpeakers) {
-        val callProfile = _callProfile.value
+        val callMetadataProfile = _callMetadataProfile.value
 
-        callProfile.calls[conversationId]?.let { call ->
+        callMetadataProfile.data[conversationId]?.let { call ->
             callingLogger.i("updateActiveSpeakers() - conversationId: $conversationId with size of: ${activeSpeakers.activeSpeakers.size}")
 
             val updatedParticipants = callMapper.activeSpeakerMapper.mapParticipantsActiveSpeaker(
                 participants = call.participants,
                 activeSpeakers = activeSpeakers
             )
 
-            val updatedCalls = callProfile.calls.toMutableMap().apply {
+            val updatedCallMetadata = callMetadataProfile.data.toMutableMap().apply {
                 this[conversationId] = call.copy(
                     participants = updatedParticipants,
                     maxParticipants = max(call.maxParticipants, updatedParticipants.size + 1)
                 )
             }
 
-            _callProfile.value = callProfile.copy(
-                calls = updatedCalls
+            _callMetadataProfile.value = callMetadataProfile.copy(
+                data = updatedCallMetadata
             )
         }
     }
 
-    private suspend fun persistMissedCallMessageIfNeeded(call: Call) {
-        if ((call.status == CallStatus.CLOSED && call.establishedTime == null) || call.status == CallStatus.MISSED) {
-            messageRepository.persistMessage(
-                Message.System(
-                    uuid4().toString(),
-                    MessageContent.MissedCall,
-                    call.conversationId,
-                    timeParser.currentTimeStamp(),
-                    call.callerId.toUserId(),
-                    Message.Status.SENT,
-                    Message.Visibility.VISIBLE
-                )
+    /**
+     * To be used only in Debug mode
+     */
+    override suspend fun deleteAllCalls() {
+        callDAO.deleteAllCalls()
```

Comment:
Usually in the repository layer, we are using `wrapStorageRequest` to handle DB operations and have error handling for free. Would be a nice to have, so we can handle possible errors in the standard way 🙌